### ISSUE TITLE
Reimplementation of the iframe (Research/implementation)

### ISSUE THREAD

**Issue by pavelsinelnikov** [SenecaCollegeBTSProjects/Group_07#13]
As of now, the user is able to click on the article, but the unreliability/slowness of each article makes the user experience negative. I wish to investigate fetching only the relevant text and images from the article. I will have a prototype of the idea completed by the end of the iteration.

**Comment by marceljar** [comment]
Agreed. Provide a write up of your efforts. Or post comments as you go through.

**Comment by pavelsinelnikov** [comment]
565c3547dc4e8fcdd738b1a1cc075c6bd47f3fdc

Replaced iframe by scraping static article page data and displaying the data as HTML in the page. This implementation only works for a limited number of sites as they are all designed differently. This also only works with static site data as dynamically rendered site data will not get retrieved in time. A link will be provided if the site does not properly provide the article page data.

An issue that I have encountered was that the cors implementation of the sites did not allow cross-site access of HTML data. More specifically, GET requests external to the site are not allowed (which is this current implementation way of retrieving data). In this case, I have enabled cors using an extension in Firefox which allows me to pull site data, bypassing these rules.

For the future implementations I would suggest using [nightmare.js](http://www.nightmarejs.org/). This allows headless browser access and does not rely on cors and bypasses the limitation of not being able to pull data from dynamically rendered sites, since it is essentially a detached browser. Of course, with all web scraping, the results may also be inaccurate and may be different on a page-to-page basis.
